turbo/src/main/kotlin/dev/hotwire/turbo/session/TurboSession.kt

```diff
@@ -748,32 +762,52 @@ class TurboSession internal constructor(
             super.onReceivedError(view, request, error)
 
             if (request.isForMainFrame && isFeatureSupported(WEB_RESOURCE_ERROR_GET_CODE)) {
-                logEvent("onReceivedError", "errorCode" to error.errorCode)
+                val visitError = TurboVisitError(
+                    type = TurboVisitErrorType.WEB_RESOURCE_ERROR,
+                    code = error.errorCode,
+                    description = if (isFeatureSupported(WEB_RESOURCE_ERROR_GET_DESCRIPTION)) {
+                        error.description.toString()
+                    } else {
+                        null
+                    }
+                )
+
+                logEvent("onReceivedError", "error" to visitError)
                 reset()
-                callback { it.onReceivedError(error.errorCode) }
+                callback { it.onReceivedError(visitError) }
             }
         }
 
         override fun onReceivedHttpError(view: WebView, request: WebResourceRequest, errorResponse: WebResourceResponse) {
             super.onReceivedHttpError(view, request, errorResponse)
 
             if (request.isForMainFrame) {
-                logEvent("onReceivedHttpError", "statusCode" to errorResponse.statusCode)
+                val visitError = TurboVisitError(
+                    type = TurboVisitErrorType.HTTP_ERROR,
+                    code = errorResponse.statusCode,
+                    description = errorResponse.reasonPhrase
+                )
+
+                logEvent("onReceivedHttpError", "error" to visitError)
                 reset()
-                callback { it.onReceivedError(errorResponse.statusCode) }
+                callback { it.onReceivedError(visitError) }
             }
         }
 
         override fun onReceivedSslError(view: WebView, handler: SslErrorHandler, error: SslError) {
             super.onReceivedSslError(view, handler, error)
             handler.cancel()
 
-            logEvent("onReceivedSslError", "url" to error.url)
+            val visitError = TurboVisitError(
+                type = TurboVisitErrorType.WEB_SSL_ERROR,
+                code = error.primaryError
+            )
+
+            logEvent("onReceivedSslError", "error" to visitError)
             reset()
-            callback { it.onReceivedError(-1) }
+            callback { it.onReceivedError(visitError) }
         }
 
-        @TargetApi(Build.VERSION_CODES.O)
```

Comment:
I noticed that this is no longer necessary, now that the library requires API `26`+